admin/views/class-yoast-input-select.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Class for generating a html select.
+ */
+class Yoast_Input_Select {
+
+	/**
+	 * @var array
+	 */
+	private $select_attributes = array();
+
+	/**
+	 * @var array Array with the options to parse.
+	 */
+	private $select_options;
+
+	/**
+	 * @var string The current selected option.
+	 */
+	private $selected_option;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param array  $select_attributes Attributes for the select, should contains: id, name and class.
+	 * @param array  $select_options    Array with the options to parse.
+	 * @param string $selected_option   The current selected option.
+	 */
+	public function __construct( array $select_attributes, array $select_options, $selected_option ) {
+
+		$this->validate_attributes( $select_attributes );
+
+		$this->select_attributes = $select_attributes;
+		$this->select_options    = $select_options;
+		$this->selected_option   = $selected_option;
+	}
+
+	/**
+	 * Print the rendered view.
+	 */
+	public function output_html() {
+		echo $this->get_html();
+	}
+
+	/**
+	 * Return the rendered view
+	 *
+	 * @return string
+	 */
+	public function get_html() {
```

Comment:
Instead of calling this function when requesting `output_html` and always using the expensive output caching we should wrap the output_html with output caching when the output is needed as return value.